chapters/en/slides/module7/module7_02.Rmd

```diff
@@ -0,0 +1,342 @@
+
+points & bars
+```
+
+Notes:
+To add two charts together,
+we can use what we learned previously about Altair's layout operators.
+To concatenate the charts vertically,
+we're here using the `&` operator.
+
+In contrast to the two scatterplots in an earlier slide,
+these charts are not created from the same base chart,
+so there selections are not linked.
+
+What if we want to link the selections
+so that when we select a region of origin in the bar chart
+the points in the scatter plot that are from the same origin would be highlighted?
+
+This would be useful to study observations from each Origin separately,
+so how can we perform this linkage in Altair?
+
+---
+
+## Linking selections from separate charts requires explicit columns to select on
+
+```{python}
+click = alt.selection_multi(fields=['Origin'])
+bars = bars.add_selection(click)
+points = points.encode(opacity=alt.condition(click, alt.value(0.9), alt.value(0.2)))
+
+points & bars
+```
+
+Notes:
+To link separate charts together,
+we need to be explicit about what happens when a selection is made.
+
+For example,
+to link a click of a bar 
+to a selection in the scatterplot,
+we need to say which column in the dataframe should be used to select on.
+
+In this case,
+we only have a single column that we use in the chart `"Origin"`,
+so this is both our only option
+and what we want since both the barchart and scatterplot
+contain this column as an encoding in the chart.
+
+To successfully link this selection
+we need to do three things:
+
+1. Create a new selection that specifies which columns should be used for selections (the `fields` parameter).
+2. Add this new selection to the bar plot
+3. Add an encoding to the scatterplot that depends on the new `click` selections. Here we use `opacity`.
+
+Now that we click the bars
+the opacity of the corresponding points changes
+as we would expect!
+
+Note that we can still click and drag in the scatterplot
+since our `brush` selection is unchanged from previously.
+
+---
+
+## A selection can be bound to the legend instead of a particular chart
+
+```{python}
+click_legend = alt.selection_multi(fields=['Origin'], bind='legend')
+points = points.encode(opacity=alt.condition(click_legend, alt.value(0.9), alt.value(0.2)))
+
+points.add_selection(click_legend)
+```
+
+Notes:
+We don't always have a barchart to drive the selections from.
+What if we only created a scatterplot,
+but still wanted to be able to select data from different Origins?
+
+Instead of using a separate chart,
+we could use the legend to make the selections.
+We can specify to Altair that we want to make the legend interactive,
+by binding `selection_multi` to it.
+
+We then need to link this new selection
+(called `click_legend` to distinguish it from the previous slide)
+to the points opacity encoding and add it to the plot.
+
+Now if you click the legend,
+it will select the corresponding observations in the scatterplot!
```

Comment:
```suggestion
We don't always have a bar chart to drive the selections from.
What if we only create a scatterplot,
but still want to be able to select data from different origins?

Instead of using a separate chart,
we could use the legend to make the selections.

We can tell Altair that we want to make the legend interactive,
by binding `selection_multi` to it.

We then need to link this new selection
to the points opacity encoding and add it to the plot.

Now if you click the legend,
it will select the corresponding observations in the scatterplot!
```